tests/pylint/test_imports.py

```diff
@@ -162,23 +181,45 @@ def test_good_root_import(
 
 
 @pytest.mark.parametrize(
-    "import_node",
+    "import_node,module_name",
     [
-        "import homeassistant.components.climate.const as climate",
-        "from homeassistant.components.climate import const",
-        "from homeassistant.components.climate.const import ClimateEntityFeature",
+        (
+            "import homeassistant.components.climate.const as climate",
+            "homeassistant.components.pylint_test.climate",
+        ),
+        (
+            "from homeassistant.components.climate import const",
+            "homeassistant.components.pylint_test.climate",
+        ),
+        (
+            "from homeassistant.components.climate.const import ClimateEntityFeature",
+            "homeassistant.components.pylint_test.climate",
+        ),
+        (
+            "from homeassistant.components.climate import const",
+            "tests.components.pylint_test.climate",
+        ),
+        (
+            "from homeassistant.components.climate.const import CONSTANT",
+            "tests.components.pylint_test.climate",
+        ),
+        (
+            "import homeassistant.components.climate.const as climate",
+            "tests.components.pylint_test.climate",
+        ),
```

Comment:
Should these be allowed?
The patch proposed on discord by @mib1185 allows these, but IMO we should only allow it when the component of the imported constants matches the component being tested.

Reply:
This should be allowed (the current pylint plugin in incorrect):
https://github.com/home-assistant/core/blob/aa0cbf0afefedafa5266b78261909c543e678a40/tests/components/advantage_air/test_climate.py#L10-L13

but this should not be allowed (the current pylint plugin is correct):
https://github.com/home-assistant/core/blob/aa0cbf0afefedafa5266b78261909c543e678a40/tests/components/advantage_air/test_climate.py#L14-L23

Reply:
Agree the latter should reject.